x/gamm/pool-models/stableswap/README.md

```diff
@@ -1,4 +1,137 @@
+$g(x_0, y_0, v, w) = k = g(x_0 + a, y_0 - b, v, w)$. As $k$ is linearly related to $v$, and $v$ is unchanged throughout the swap, we can simplify the equation to be reasoning about $k' = \frac{k}{v}$ as the constant, and $h$ instead of $g$
+
+We then model the solution by finding a function $\text{solve\_cfmm}(x, w, k') = y\text{ s.t. }h(x, y, w) = k'$.
+Then we can solve the swap amount out by first computing $k'$ as $k' = h(x_0, y_0, w)$, and 
+computing $y_f := \text{solve\_cfmm}(x_0 + a, w, k')$. We then get that $b = y_0 - y_f$.
+
+So all we need is an equation for $\text{solve\_cfmm}$! Its essentially inverting a multi-variate polynomial, and in this case is solvable: [wolfram alpha link](https://www.wolframalpha.com/input?i=solve+for+y+in+x+*+y+*+%28x%5E2+%2B+y%5E2+%2B+w%29+%3D+k)
+
+Or if were clever with simplification in the two asset case, we can reduce it to: [desmos link](https://www.desmos.com/calculator/ktdvu7tdxv).
+
+These functions are a bit complex, which is fine as they are easy to prove correct. However, they are relatively expensive to compute, the latter needs precision on the order of x^4, and requires computing multiple cubic roots.
+
+Instead there is a more generic way to compute these, which we detail in the next subsection.
+
+#### Iterative search solution
+
+Instead of using the direct solution for $\text{solve\_cfmm}(x, w, k')$, instead notice that $h(x, y, w)$ is an increasing function in $y$. 
+So we can simply binary search for $y$ such that $h(x, y, w) = k'$, and we are guaranteed convergence within some error bound. 
+
+In order to do a binary search, we need bounds on $y$. The lowest lowerbound is $0$, and the largest upperbound is $\infty$. The maximal upperbound is obviously unworkable, and in general binary searching around wide ranges is unfortunate, as we expect most trades to be centered around $y_0$. This would suggest that we should do something smarter to iteratively approach the right value. Notice that $h$ is super-linearly related in $y$, and at most cubically related to $y$. So $2 * h(x,y,w) <= h(x,2*y,w) <= 8 * h(x,y,w)$. We can use this fact to get a pretty-good initial upperbound and lowerbound guess for $y$. As these bounds are relatively tight (only 3 binary search steps away from one another), we do not aim to optimize further.
+
+We detail how we use this below:
+<!--Let $k_{guess,0} := h(x_f, y_0, w)$. If $k_{guess,0} > k$, then $y_0$ is our upper bound --->s
+
+```python
+def iterative_search(x_f, y_0, w, k, err_tolerance):
+  k_0 = h(x_f, y_0, w)
+  lowerbound, upperbound = y_0, y_0
+  if k_0 > k:
+    # need to find a lowerbound, assume cubic relationship
+  elif k_0 < k:
+    # need to find an upperbound, assume linear relationship
+
+def binary_search(x_f, y_0, w, k, err_tolerance):
+  iter_count = 0
+  y_est = y_0
+  k_est = h(x_f, y_0, w)
+  lowerbound = 0
+  upperbound = 
+```
+
+### Spot Price
```

Comment:
Are these considered as TODOs for future reference?

Reply:
Yeah, just things that should be filled in before stableswap has a full spec